src/__tests__/select-event.test.tsx

```diff
@@ -124,13 +127,21 @@ describe("The select event helpers", () => {
     expect(form).toHaveFormValues({ food: "papaya" });
   });
 
+  it("types in and adds a new option but does not select it by default", async () => {
+    const { form, input } = renderForm(<Creatable {...defaultProps} />);
+    expect(form).toHaveFormValues({ food: "" });
+    await selectEvent.create(input, "papaya", { autoSelect: false });
+    expect(form).toHaveFormValues({ food: "" });
+  });
+
   it("types in and adds a new option with custom create label when searching by fixed string", async () => {
     const { form, input } = renderForm(
       <Creatable {...defaultProps} formatCreateLabel={() => "Add new option"} />
     );
     expect(form).toHaveFormValues({ food: "" });
 
     await selectEvent.create(input, "papaya", {
+      autoSelect: true, 
```

Comment:
Since `autoSelect` already defaults to true, can we remove this line?

Reply:
What I think is going on here, is that `config` is being overwritten and the value for `config` only sets `createOptionText`.  A more longform way of thinking of this:

```
const config: CreateConfig = {
  createOptionText: "Add new option",
}

console.log(JSON.stringify(config));

await selectEvent.create(input, "papaya", config);
```

This will print the following `{"createOptionText":"Add new option"}`...there is no `autoSelect` key in that object.  If I remove `autoSelect: true`, the test will fail.  So `autoSelect` defaults to `true`, unless you provide a value for `config`, then `autoSelect` will have whatever value the `config` parameter has for `autoSelect`.

Disclaimer: I'm not a JS/TS expert, so I may simply be doing something wrong 😁

Reply:
Oh I see!

The syntax is pretty weird, but to make the `autoSelect` option really optional, you can write line 104 like this:

```ts
   { autoSelect = true, ...config }: CreateConfig = {} 
```

(and then directly use `autoSelect` instead of `config.autoSelect` line 112)

Reply:
Make sense, learned something new today!